hibernate-envers/src/main/java/org/hibernate/envers/configuration/internal/metadata/MetadataTools.java

```diff
@@ -285,38 +297,68 @@ public static Element createJoin(
 		return joinMapping;
 	}
 
-	public static void addColumns(Element anyMapping, Iterator selectables) {
+	public static void addColumns(Element anyMapping, Iterator<?> selectables, Metadata metadata) {
+		addColumns( anyMapping, selectables, metadata, metadata.getDatabase().getDialect() );
+	}
+
+	public static void addColumns(Element anyMapping, Iterator<?> selectables, Mapping mapping, Dialect dialect) {
 		while ( selectables.hasNext() ) {
 			final Selectable selectable = (Selectable) selectables.next();
 			if ( selectable.isFormula() ) {
 				throw new FormulaNotSupportedException();
```

Comment:
I checked the usage of Selectable and found there is a hidden logic that if `isFormula` is false, it would be of `Column` type (including subtype), but this seems a little bit weird and deciding whether it is `Column` or not would be more straightforward. Ignore my message if my concern is invalid for knowledge gap reason.

Reply:
We actually use the `#isFormula()` pattern in quite a number of places across the code base to differentiate between whether a `Selectable` is a formula or column sub-type.  I don't believe there is any reason to change whats here unless I'm misunderstanding your point here.

Reply:
You didn't misunderstand my point. Thanks for the explanation.